proposals/VK_EXT_map_memory2.asciidoc

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 The Khronos Group, Inc.
+//
+// SPDX-License-Identifier: CC-BY-4.0
+
+# VK_EXT_map_memory2
+:toc: left
+:refpage: https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/
+:sectnums:
+
+This document proposes adding extensible version of `vkMapMemory()` and
+`vkUnmapMemory()`.
+
+## Problem Statement
+
+The current Vulkan memory mapping entrypoints are not really extensible.
+`vkMapMemory()` does have a flags argument which is currently unused but
+neither `vkMapMemory()` nor `vkUnmapMemory()` take an input struct with a
+`pNext` which can be extended.
+
+## Proposal
+
+Add new `vkMapMemory2EXT()` and `vkUnmapMemory2EXT()` entrypoints which
+take input structs which are extensible via the usual `pNext` mechanism:
+[source,c]
+----
+typedef struct VkMemoryMapInfoEXT {
+    VkStructureType     sType;
+    const void*         pNext;
+    VkDeviceMemory      memory;
+    VkDeviceSize        offset;
+    VkDeviceSize        size;
+    VkMemoryMapFlags    flags;
+} VkMemoryMapInfoEXT;
+
+VKAPI_ATTR VkResult VKAPI_CALL vkMapMemory2EXT(
+    VkDevice                                    device,
+    const VkMemoryMapInfoEXT*                   pMemoryMapInfo,
+    void**                                      ppData);
+
+typedef struct VkMemoryUnmapInfoEXT {
+    VkStructureType          sType;
+    const void*              pNext;
+    VkDeviceMemory           memory;
+    VkMemoryUnmapFlagsEXT    flags;
+} VkMemoryUnmapInfoEXT;
```

Comment:
Interaction question here that probably needs spelling out (at least in the map_placed proposal) - what happens if you map separate multiple ranges via memoryMapRangePlaced? Language you've got so far says that it unmaps _the_ range, which could plausibly refer to the range of `memory`, which is fine, but firstly I want to double check that, and secondly - should there be a way to unmap a subrange in the same way?

Reply:
You're not allowed to map two different ranges.  This VU still holds:
> `VUID-vkMapMemory-memory-00678`
>
> `memory` **must** not be currently host mapped

It unmaps the mapped range.  There's only one mapped range.

If we wanted to allow more direct control of memory mappings, that feels like a different feature.  I don't see why it wouldn't be supportable on Linux but I also don't see any real value.  Not supportable on Windows, of course.

Reply:
Ah ok, didn't realise you could only have a single mapped range full stop. I suspect it might be useful with SVM-like features in the future, but on its own right now agree it's pretty useless. Fine with this resolution then.